user_interface/EbikeUIController/EbikeUIController.ino

```diff
@@ -0,0 +1,720 @@
+    changeActiveRegenRequest();
+    Serial.println("Request to change the active regenerative braking method has been sent to Motor Controller via CAN");
+
+    // Update the display to show the new regenerative braking method
+    updateRegen();
+
+  }
+
+  // Update the display to show the latest values
+  updateDisplay();
+  
+}
+
+// Function to update the current speed
+void adjustSpeedRequest(int adjustment) {
+
+  // Code to send request to motor controller FW to adjust the current speed goes here
+
+  if (adjustment == 1) {
+    // Send increase speed request to motor controller
+
+    if (throttle == 100) {
+      return;
+    } else {
+      throttle += 5;
+    }
+
+  } else if (adjustment == -1) {
+    // Send decrease speed request to motor controller
+    
+    if (throttle == 0) {
+      return;
+    } else {
+      throttle -= 5;
+    }
+  
+  }
+
+}
+
+// Function to update the display variable that shows the current speed
+void updateSpeed(void) {
+
+  // Code to read the current speed signal from the motor controller and update the speed variable goes here
+
+  // This is connected to a potentiometer for testing purposes only.
+  speed = map(analogRead(14), 0, 1023, 0, 99);
+  throttle = map(analogRead(14), 0, 1023, 0, 99);
+
+}
+
+// Function to update the display variable that shows the current power output
+void updatePower(void) {  
+
+  // Code to read the current motor power output signal from the motor controller and update the power variable goes here
+  
+  // This is connected to a potentiometer for testing purposes only.
+  power = map(analogRead(14), 0, 1023, 0, 999);
+
+}
+
+// Function to update the display variable that shows the current motor temperature
+void updateTemp(void) {
+
+  // Code to read the current motor temperature signal from the motor controller and update the temperature variable goes here
+  
+  // This is connected to a potentiometer for testing purposes only.
+  temp = map(analogRead(14), 0, 1023, 0, 99);
+
+}
+
+// Function to update the display variable that shows the current battery range
+void updateBatteryRange(void) {
+
+  // Code to read the current battery range signal from the BMS and update the battery range variable goes here
+
+  // This is connected to a potentiometer for testing purposes only.
+  batteryRange = map(analogRead(14), 0, 1023, 0, 99);
+
+}
+
+// Function to update the active regenerative braking method
+void changeActiveRegenRequest(void) {
+
+  // Code to send request to motor controller FW to toggle the active regenerative braking method goes here
+
+}
+
+// Function to update the display variable that shows the active regenerative braking method
+void updateRegen(void) {
+
+  // Code to update the active regenerative braking method variable goes here
+  if (activeRegen == 1) {
+    activeRegen = 2;
+  } else {
+    activeRegen = 1;
+  }
+
+}
+
+// Function to update the display with the latest values
+void updateDisplay(void) {
```

Comment:
Again, would be really nice to have this separated into a display library/class in a separate file. The UI parameters (speed, throttle, power etc.) could be packaged into a struct to make it easier to pass.

Reply:
Marked for future steps.